### ISSUE TITLE
WebAuthenticationBroker from ViewModel?

### ISSUE THREAD

**Issue by weitzhandler** [MicrosoftDocs/winrt-api#1782]
Hi,

I wonder if there is any documentation on how to handle authentication identity, token, claims, and user, from when the UI logic is decoupled into a ViewModel etc. layer that's UI-agnostic.

Related: https://github.com/microsoft/ProjectReunion/issues/31

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 3c758cb2-ad03-f28e-6947-21337462753a
* Version Independent ID: 3e818dc7-e032-c92b-80ba-0c0773391fec
* Content: [WebAuthenticationBroker Class (Windows.Security.Authentication.Web) - Windows UWP applications](https://docs.microsoft.com/en-us/uwp/api/windows.security.authentication.web.webauthenticationbroker?view=winrt-19041)
* Content Source: [winrt-api-build/xml/Windows.Security.Authentication.Web/WebAuthenticationBroker.xml](https://github.com/MicrosoftDocs/winrt-api/blob/docs/winrt-api-build/xml/Windows.Security.Authentication.Web/WebAuthenticationBroker.xml)
* Product: **uwp**
* Technology: **winrt-reference**
* GitHub Login: @lastnameholiu
* Microsoft Alias: **alholiu**

**Comment by bgavrilMS** [comment]
@weitzhandler - I'm providing a few ideas based on my experience as the MSAL.NET developer. MSAL.NET integrates with WAB. 

WAB is deprecated and as far as I know there is no new feature work happening there. I have not investigated if there are win32 projections of the WAB API.

I would suggest instead you look at WAM, which provides much better support especially around device authentication, integration with Windows Hello etc. There are win32 projections of the WAM API. From my experience with integrating with WAM (currently [in progress](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1962)), the difference between UWP and Win32 is: 

- on Win32, you are required to parent the Authentication controls to the a window handle
- on UWP, this is not required, because you can always get the current view
- on both, some of Authentication popups require you to be on an UI thread

**Comment by weitzhandler** [comment]
Thanks @bgavrilMS for your kind and detailed reply,

> MSAL.NET

MSAL doesn't cover custom identity management or legacy authentication with username & password against a private authentication provider (https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1538).

> I would suggest instead you look at WAM

Sorry I mixed between WAB and WAM.

Anyway, as I commented [here](https://github.com/microsoft/ProjectReunion/issues/31#issuecomment-678688410), I wish there was a lower level agnostic abstraction of client-side authentication flow and token management that can be managed in the ViewModel side as well.
Currently it all has to be hand-written down, hence my request for an example for using this with a real life example involving MVVM and UI decoupling.
